src/main/java/com/minecolonies/coremod/util/BlockPosUtil.java

```diff
@@ -127,6 +129,57 @@ public static BlockPos readFromByteBuf(@NotNull final ByteBuf buf)
     }
 
     /**
+     * this checks that you are not in liquid.  Will check for all liquids, even those from other mods
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @param sender uses the player to get the world
+     * @return isSafe true=safe false=water or lava
+     */
+    public static boolean isPositionSafe(@NotNull ICommandSender sender, BlockPos blockPos)
+    {
+        return sender.getEntityWorld().getBlockState(blockPos).getBlock() != Blocks.AIR
+                && !sender.getEntityWorld().getBlockState(blockPos).getMaterial().isLiquid()
+                && !sender.getEntityWorld().getBlockState(blockPos.up()).getMaterial().isLiquid();
+    }
+
+    /**
+     * this checks that you are not in the air or underground.
+     * If so it will look up and down for a good landing spot before TP.
+     *
+     * @param blockPos for the current block LOC.
+     * @return blockPos to be used for the TP.
+     */
+    public static BlockPos findLand(final BlockPos blockPos, final World world)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document the parameter(s): world [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)